includes/razorpay-subscriptions.php

```diff
@@ -104,62 +104,58 @@ protected function getSubscriptionCreateData($orderId)
 
         $length = (int) WC_Subscriptions_Product::get_length($product['product_id']);
 
-        $renewalDate = WC_Subscriptions_Product::get_first_renewal_payment_time($product['product_id']);
-
         //The first payment is always set as an upfront amount to support woocommerce discounts like fixed
-        // cart discounts which is only for the first payment. The start_date of subscription would be next
-        //renewal date
-
-        if ($length ==1)
-        {
-            $subscriptionData = array(
-                'customer_id'     => $customerId,
-                'plan_id'         => $planId,
-                'quantity'        => (int) $product['qty'],
-                'total_count'     => $length,
-                'customer_notify' => 0,
-                'notes'           => array(
-                    'woocommerce_order_id'   => $orderId,
-                    'woocommerce_product_id' => $product['product_id']
-                ),
-            );
-
-            return $subscriptionData;
-        }
+        // cart discounts which is only for the first payment.
 
         $subscriptionData = array(
-            // TODO: Doesn't work with trial periods currently
             'customer_id'     => $customerId,
             'plan_id'         => $planId,
             'quantity'        => (int) $product['qty'],
-            'total_count'     => $length -1,
-            'start_at'        => $renewalDate,
+            'total_count'     => $length,
             'customer_notify' => 0,
             'notes'           => array(
                 'woocommerce_order_id'   => $orderId,
                 'woocommerce_product_id' => $product['product_id']
             ),
         );
 
+        $signUpFee = WC_Subscriptions_Product::get_sign_up_fee($product['product_id']);
+
+        // We add the signup fee as an addon
+
+        if ($signUpFee)
+        {
+            $subscriptionData['addons'] = array(array('item' => $this->getUpfrontAmount($signUpFee, $order, $product)));
+        }
+
+        if ($length ==1)
```

Comment:
Not working properly in case of 1 month of subscription period and trail period.

Reply:
As woo-commerce support "no expiry " in subscription length time frame. We need to handle it based on Razorpay maximum length time frame in this case.

Reply:
Currently we are not supporting never expire case. So we throw an exception in case of if merchant choose never expire subscription.